qualtran/_infra/registers.py

```diff
@@ -143,15 +143,18 @@ def __init__(self, registers: Iterable[Register]):
         self._rights = _dedupe((reg.name, reg) for reg in self._registers if reg.side & Side.RIGHT)
 
     @classmethod
-    def build(cls, **registers: Union[int, sympy.Expr]) -> 'Signature':
+    def build(cls, ignore_zero: bool = True, **registers: Union[int, sympy.Expr]) -> 'Signature':
```

Comment:
Adding an `ignore_zero:` flag to make tests pass for this PR would be hacky -- Let's pull out this change in a separate PR and fix it properly OR update the block encodings design so we don't have 0 sized registers. The only reason tests are not failing for block encodings with 0 bitsizes is because we don't have the right tests that would trigger a failure. 

I suggest it may be easier to continue the assumption that we won't have registers with bitsize 0 (and therefore not make this change). The block encodings implementation can do the following:

Iterate on `self.subbloq.registers` and for each register which is present, pull out the ancilla / resource part from `soqs` dict. This pattern has been used across the library at multiple places where the signature of the subbloq can be dynamic (for example: where the subbloq may or may not have a control register and this behavior is dependent upon an attribute `cvs` that the subbloq expects).

Reply:
@mpharrigan As we just discussed this today, do you want to weigh in on the question of zero-sized registers?

Reply:
I have removed the zero-sized registers from this PR.